packages/ingest/http-auth.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const http = require('http');
+const https = require('https');
+const request = require('request'); // eslint-disable-line node/no-extraneous-require
+const { parse } = require('url');
+
+function genUriOptions(uri, headers) {
+  return {
+    uri,
+    method: 'GET',
+    followRedirect: false,
+    headers
+  };
+}
+
+function authString() {
+  return Buffer.from(
+    `${process.env.EARTHDATA_USER}:${
```

Comment:
since this is adding support for basic auth I wonder if we can use something more generic for the username/pass env variables?